### ISSUE TITLE
LitePlacer crashes when TinyG is not connected

### ISSUE THREAD

**Issue by mrandt** [jkuusama/LitePlacer-ver2#1]
If I start LitePlacer software on my office machine (which has no TinyG connected) it crashes. The main Window does not come up and I have to kill LitePlacer.exe via Windows task manager.

**Comment by thethereza** [comment]
yeah, it will open some other com port and freak out.  ideally, the code should detect if a tinyg is on the other side - not sure how to do that currently.

**Comment by mrandt** [comment]
That seemed to work in @jkuusama original version - no idea how he solved it.

**Comment by mrandt** [comment]
I investigated this a little more. On connecting to TinyG we should do the following:
- Need to enable timeout settings for write and read on the serial port and handle the exception; in case the port does not work.
- Check if TinyG is present and responding, by issuing a command and  parsing the expected repsonse.
- If TinyG had been running before, even more synchronization might be necessary to reach a well defined state.

I started a discussion with the TinyG firmware team here:
https://groups.google.com/forum/#!topic/devtinyg/0EBqsrkXCFk

They suggested to take their "ChilliPepper" software as an example.

**Comment by thethereza** [comment]
Can u find a link to the source of that software?

On Wednesday, July 15, 2015, mrandt notifications@github.com wrote:

> I investigated this a little more. On connecting to TinyG we should do the
> following:
> - Need to enable timeout settings for write and read on the serial
>   port and handle the exception; in case the port does not work.
> - Check if TinyG is present and responding, by issuing a command and
>   parsing the expected repsonse.
> - If TinyG had been running before, even more synchronization might be
>   necessary to reach a well defined state.
> 
> I started a discussion with the TinyG firmware team here:
> https://groups.google.com/forum/#!topic/devtinyg/0EBqsrkXCFk
> 
> They suggested to take their "ChilliPepper" software as an example.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/jkuusama/LitePlacer-ver2/issues/1#issuecomment-121543133
> .

## 

-Reza

**Comment by mrandt** [comment]
I tried to find the source but could not find it right away. Not sure it is open source?

But you can easily try it out in your browser and trace the commands going back and forth between serial and hardware controller when connecting to TinyG:
http://chilipeppr.com/tinyg
